src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -408,7 +408,7 @@ public static Type GetElementType(Type propertyType, Type parentType, MemberInfo
                 }
             }
 
-            if (IsNonGenericInterface(propertyType))
+            if (IsNonGenericInterface(propertyType) || propertyType == typeof(IDictionary))
```

Comment:
Why isn't IDictionary just part of IsNonGenericInterface?  If that causes problems elsewhere, then the naming of IsNonGenericInterface seems misleading (even with the IsSupportedNonGenericInterface I'd suggested in the other PR).